packages/gatsby/src/utils/worker/render-html.ts

```diff
@@ -312,34 +312,38 @@ export const renderHTMLProd = async ({
     }
   }
 
-  await Bluebird.map(paths, async pagePath => {
-    try {
-      const pageData = await readPageData(publicDir, pagePath)
-      const resourcesForTemplate = await getResourcesForTemplate(pageData)
-
-      const { html, unsafeBuiltinsUsage } = htmlComponentRenderer.default({
-        pagePath,
-        pageData,
-        ...resourcesForTemplate,
-      })
-
-      if (unsafeBuiltinsUsage.length > 0) {
-        unsafeBuiltinsUsageByPagePath[pagePath] = unsafeBuiltinsUsage
-      }
-
-      return fs.outputFile(getPageHtmlFilePath(publicDir, pagePath), html)
-    } catch (e) {
-      if (e.unsafeBuiltinsUsage && e.unsafeBuiltinsUsage.length > 0) {
-        unsafeBuiltinsUsageByPagePath[pagePath] = e.unsafeBuiltinsUsage
-      }
-      // add some context to error so we can display more helpful message
-      e.context = {
-        path: pagePath,
-        unsafeBuiltinsUsageByPagePath,
+  await Bluebird.map(
+    paths,
+    async pagePath => {
+      try {
+        const pageData = await readPageData(publicDir, pagePath)
+        const resourcesForTemplate = await getResourcesForTemplate(pageData)
+
+        const { html, unsafeBuiltinsUsage } = htmlComponentRenderer.default({
+          pagePath,
+          pageData,
+          ...resourcesForTemplate,
+        })
+
+        if (unsafeBuiltinsUsage.length > 0) {
+          unsafeBuiltinsUsageByPagePath[pagePath] = unsafeBuiltinsUsage
+        }
+
+        return fs.outputFile(getPageHtmlFilePath(publicDir, pagePath), html)
+      } catch (e) {
+        if (e.unsafeBuiltinsUsage && e.unsafeBuiltinsUsage.length > 0) {
+          unsafeBuiltinsUsageByPagePath[pagePath] = e.unsafeBuiltinsUsage
+        }
+        // add some context to error so we can display more helpful message
+        e.context = {
+          path: pagePath,
+          unsafeBuiltinsUsageByPagePath,
+        }
+        throw e
       }
-      throw e
-    }
-  })
+    },
+    { concurrency: 2 }
```

Comment:
wouldn't we want the concurrency to match the size of the jest worker pool?

Reply:
This is concurrency within one worker. So this effectively sets total concurrency across all workers to `workerPoolSize * 2`. I was hesitating if we should set `{ concurrency: 1 }` or `{ concurrency: 2 }` here but `2` seems to keep memory usage as low as `1`, so settled on value `2`.

To clarify your question - do you mean changing it to `1` here? Because setting it to `workerPoolSize` will effectively make it `workerPoolSize * workerPoolSize` which doesn't make much sense to me unless I am missing something.

Reply:
oh my bad. I didn't read this closely and thought this was in the main node process. Yeah 2 makes sense then. SSR is pure CPU so reading in any more page-data.json files wouldn't speed up anything and 1 would very slightly slow it down as the next SSR has to wait for the read to finish.